R/git.R

```diff
@@ -120,10 +120,18 @@ git_commits <- function(branch, root = NULL) {
   commits
 }
 
+git_latest_commit <- function(root = NULL) {
+  args <- c("log", "--pretty='%h'", "-n", "1")
+  git_run(args, root = root, check = TRUE)$output
+}
 
 get_reports <- function(branch, commit, root) {
-  if (branch == "master") {
-    ## Get all reports in commit if on master branch
+  if (identical(branch, "master") || is.null(branch)) {
```

Comment:
Could move to 2 endpoints
* Get unmerged reports taking a commit hash, returns list of all reports which have modications on that commit which have not been merged into master
* Get reports which takes a commit hash or NULL and returns the list of all reports available in that commit
It will then be up to web app to call the 2nd endpoint on production when on production/user selects master branch

Moving to 2 endpoints puts that bit of logic into web app instead of all the requirements being contained in orderly.server - maybe that is OK though?

Reply:
Purely from an OW perspective, I would expect:
- `git_branches` to return only branches containing unmerged commits (i.e. commits that aren't on master)
- `get_commits` to return for the the relevant branch only commits that aren't on master
- `get_reports` to accept _either_ no branch or commit (where git has previously been indicated to be disabled) _or_ both a branch and commit (where the branch is _either_ master _or_ the commit is one that hasn't been merged into master)

Reply:
So get reports can either be called
1. With no branch or commit (this is from e.g. production where `git_supported` is FALSE)
1. With a branch other than master & unmerged commit (e.g. on a feature branch on science)
1. With master branch and a commit (e.g. on science when a user selects "master")

In these cases we want to list
1. All reports in src dir on latest commit on master branch
1. All reports which have changes in them from this commit compared with master branch
1. All reports in src dir on this commit

If we have 1 endpoint it will need to switch behaviour on master or null branch vs non-master branch. If we split it so these 2 conditions are handled separately then orderly web will have to manage that logic about which endpoint to call when branch it encounters is `master` which I think is not ideal. Spoke to Mark also about splitting 2 endpoints 1 which deals with no arg case (1) and one which deals with branch & commit args (2 & 3) but then this 2nd endpoint would still need to switch between the two cases.

Reply:
Have updated `spec.md` to details about new endpoints